src/resources/views/crud/fields/select_multiple.blade.php

```diff
@@ -18,7 +21,7 @@ class="form-control"
         @include('crud::fields.inc.attributes')
     	multiple>
 
-		@if (!isset($field['allows_null']) || $field['allows_null'])
```

Comment:
In this case, if `allows_null` was not set, it was true by default. Now if it's not set, it will be the nullable value on the database, is this a breaking change? I really don't know 😅 just asking ...

I think it's not a breaking change because for the developer crud to work, column must have been nullable already on the database.

So I think this is safe, @pxpm please just double check my though ahaha

Reply:
Now that you talk about it it might be a breaking change. 

Before, if user did not specify nothing in `allows_null` and have a `non-nullable` database column, the field would allow null by default (render with the placeholder). 
With this change by default it would `not allow null` (because the column is not nullable in database), so rendering the field differently (when you open the form, an entry is automatically selected for you). To have the same behaviour you would need to specify `allows_null => true`. 

I think that this was the only field that was not using `isColumnNullable`. It's unfortunate.

We could push the changes without this field and create a PR to merge on 4.2 with this BC or merge this whole PR in 4.2.

@tabacitu what do you think ?

Reply:
Previously:
| field allows_null | db column nullable | result (allows_null?) |
| - | - | - |
| ✔ | ✔ | ✔ |
| ✔ | ❌ | ✔ |
| ❌ | ✔ | ❌ |
| ❌ | ❌ | ❌ |
| _unset_ | ✔ | ✔ |
| _unset_ | ❌ | ✔ |

With this PR:
| field allows_null | db column nullable | result (allows_null?) |
| - | - | - |
| ✔ | ✔ | ✔ |
| ✔ | ❌ | ✔ |
| ❌ | ✔ | ❌ |
| ❌ | ❌ | ❌ |
| _unset_ | ✔ | ✔ |
| _unset_ | ❌ | ❌ _(problem)_ |

Had to create this table to properly understand what's going on 😅

So the problem is, when developers don't set allows_null, even if the db column is nullable, now it shows the empty option `-`, when it shouldn't right?

@pxpm isn't this more like a bug we have now?
Because if the developer don't set `field: required` on the request, it will throw one of those huge errors; `Column 'name' can't be empty`.

Reply:
Indeed, that's the problem I described before, it will have a different behaviour than in current version. It will not make the fiield stop working, it will just render differently. Not sure if this is a BC "per se".

@tabacitu

Reply:
Hmm... good catch!  Yeah... I think it's technically a bug right now, it probably shouldn't default to `true` unless the column is `nullable`. But then again, who knows how devs might have used this. So let's play it safe. Let's consider this a _breaking change_ and do this:
- [ ] inside this PR, make it default to `true` like it was before (no change in behaviour, just refactoring, so we can merge this PR into 4.1);
- [ ] create a different PR, starting from this branch, that just changes the default to be `$crud->model::isColumnNullable($field['name'])`, which is the EXPECTED behaviour, but unfortunately it'd be a breaking change, so we'll merge that in 4.2 and include it in the upgrade guide and everything;

That way:
- we can still merge this PR in 4.1;
- we can document the change in behaviour for 4.2;

I think it's a good compromise right?

Reply:
Like we discussed I think it's the "safest" approach.

I'v just submited the PR that allows this one to be merged into 4.1.

Will create the one with BC for 4.2.

Best,
Pedro

Reply:
Is this ready to merge? Do we have a 4.2 breaking one to complete this?

Reply:
@tabacitu I talked to @pxpm, the PR is here; https://github.com/Laravel-Backpack/CRUD/pull/3419.
I created the branch based on this branch, so it carries the changes on this branch, but I will be fixed when this is merged to master I think.